operator/controllers/csiscaleoperator_controller.go

```diff
@@ -542,7 +549,9 @@ func (r *CSIScaleOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			},
 			UpdateFunc: func(e event.UpdateEvent) bool {
 				if isCSIResource(e.ObjectNew.GetName(), resourceKind) {
-					if !reflect.DeepEqual(e.ObjectOld.(*corev1.Secret).Data, e.ObjectNew.(*corev1.Secret).Data) {
+					if resourceKind == corev1.ResourceConfigMaps.String() {
+						r.restartDriverPods(mgr, "updated", resourceKind, e.ObjectOld.GetName())
```

Comment:
Please add a condition to check for configmap data change here, otherwise change in metadata fields also causes driver pod restarts

Reply:
Addessed!